src/main.c

```diff
@@ -35,6 +35,41 @@ int chunked(float x) {
     return floorf(roundf(x) / CHUNK_SIZE);
 }
 
+int is_connected() {
+    if (g->server_addr[0] == '\0'
+        || g->server_user[0] == '\0'
+        || g->server_pass[0] == '\0')
+        return 0;
+    return 1;
+}
+
+void connect_console_command(char *str) {
+    if (g->server_addr[0] == '\0') {
+        if (!check_server(str)) {
+            snprintf(g->text_message, 1024,
+                "Failed to resolve %s, try again.\n", str);
+            return;
+        }
+        strncpy(g->server_addr, str, MAX_ADDR_LENGTH);
+        snprintf(g->text_message, 1024, "Select a new/existing username");
+        snprintf(g->text_prompt, 1024, "User");
+        g->typing = 1;
+        g->typing_buffer[0] = '\0';
+    } else if (g->server_user[0] == '\0') {
+        strncpy(g->server_user, str, 32);
+        snprintf(g->text_message, 1024, "Enter a password to login/create account");
+        snprintf(g->text_prompt, 1024, "Password");
+        g->typing = 1;
+        g->typing_buffer[0] = '\0';
+    } else if (g->server_pass[0] == '\0') {
+        strncpy(g->server_pass, str, 32);
+        snprintf(g->text_message, 1024, "Enter server %s as %s", g->server_addr, g->server_user);
```

Comment:
@petterarvidsson yeah, agree ... I started to get tired to I took a little shortcut :)